packages/integer-field/src/vaadin-integer-field.js

```diff
@@ -0,0 +1,175 @@
+       * - `keyDown` events if the entered key doesn't match `/^_enabledCharPattern$/`
+       * - `paste` events if the pasted text doesn't match `/^_enabledCharPattern*$/`
+       * - `drop` events if the dropped text doesn't match `/^_enabledCharPattern*$/`
+       *
+       * For example, to enable entering only numbers and minus signs,
+       * `_enabledCharPattern = "[\\d-]"`
+       * @protected
+       */
+      _enabledCharPattern: {
+        type: String,
+        value: '[-+\\d]',
+        observer: '_enabledCharPatternChanged'
+      }
+    };
+  }
+
+  constructor() {
+    super();
+
+    this._boundOnPaste = this._onPaste.bind(this);
+    this._boundOnDrop = this._onDrop.bind(this);
+    this._boundOnBeforeInput = this._onBeforeInput.bind(this);
+  }
+
+  /**
+   * @param {HTMLElement} node
+   * @protected
+   */
+  _addInputListeners(node) {
+    super._addInputListeners(node);
+
+    node.addEventListener('paste', this._boundOnPaste);
+    node.addEventListener('drop', this._boundOnDrop);
+    node.addEventListener('beforeinput', this._boundOnBeforeInput);
+  }
+
+  /**
+   * @param {HTMLElement} node
+   * @protected
+   */
+  _removeInputListeners(node) {
+    super._removeInputListeners(node);
+
+    node.removeEventListener('paste', this._boundOnPaste);
+    node.removeEventListener('drop', this._boundOnDrop);
+    node.removeEventListener('beforeinput', this._boundOnBeforeInput);
+  }
+
+  /**
+   * Override an event listener from `ClearButtonMixin`
+   * to avoid adding a separate listener.
+   * @param {!KeyboardEvent} event
+   * @protected
+   */
+  _onKeyDown(event) {
+    if (this._enabledCharPattern && !this.__shouldAcceptKey(event)) {
+      event.preventDefault();
+    }
+
+    super._onKeyDown(event);
+  }
+
+  /** @private */
+  __shouldAcceptKey(event) {
+    return (
+      event.metaKey ||
+      event.ctrlKey ||
+      !event.key || // allow typing anything if event.key is not supported
+      event.key.length !== 1 || // allow "Backspace", "ArrowLeft" etc.
+      this.__enabledCharRegExp.test(event.key)
+    );
+  }
+
+  /** @private */
+  _onPaste(e) {
+    if (this._enabledCharPattern) {
+      const pastedText = (e.clipboardData || window.clipboardData).getData('text');
+      if (!this.__enabledTextRegExp.test(pastedText)) {
+        e.preventDefault();
+      }
+    }
+  }
+
+  /** @private */
+  _onDrop(e) {
+    if (this._enabledCharPattern) {
+      const draggedText = e.dataTransfer.getData('text');
+      if (!this.__enabledTextRegExp.test(draggedText)) {
+        e.preventDefault();
+      }
+    }
+  }
+
+  /** @private */
+  _onBeforeInput(e) {
+    // The `beforeinput` event covers all the cases for `_enabledCharPattern`: keyboard, pasting and dropping,
+    // but it is still experimental technology so we can't rely on it. It's used here just as an additional check,
+    // because it seems to be the only way to detect and prevent specific keys on mobile devices. See issue #429.
```

Comment:
I would suggest adding a link to the issue in this comment rather than just an ID.

Does it mean this issue: https://github.com/vaadin/web-components/issues/429?

The issue following from the above link looks quite irrelevant to what is described in the comment...

Reply:
Good catch, added a link to the original issue from the `vaadin-text-field` repo.